desks-front/src/components/UI/Desk/DeskContainer/DeskContainer.jsx

```diff
@@ -1,89 +1,87 @@
 import { useEffect, useState } from "react";
-
 import { Button, ButtonToolbar, OverlayTrigger } from "react-bootstrap";
 import "./DeskContainer.css";
 import { AvailableTooltip } from "../../Tooltip/Tooltips";
 import DropdownMenu from "../../DropdownMenu";
 import SelectedDeskDisplay from "../SelectedDeskDisplay";
-
-// temporary data, should be fetched from firebase
-const desks = [
-    { location: "hd", room: 1, desk: 1, addedBy: "admin@admin.com" },
-    { location: "ns", room: 1, desk: 1, addedBy: "admin@admin.com" },
-    { location: "ns", room: 1, desk: 5, addedBy: "admin@admin.com" },
-    { location: "ns", room: 1, desk: 2, addedBy: "admin@admin.com" },
-    { location: "md", room: 1, desk: 5, addedBy: "admin@admin.com" },
-    { location: "md", room: 2, desk: 2, addedBy: "admin@admin.com" },
-];
+import getDesksByLocationFromDb from "../../../../utils/db/getDesksByLocationFromDb";
 
 const menuOptions = [
-    { value: "", label: "- choose option -" },
-    { value: "hd", label: "Heidelberg" },
-    { value: "md", label: "Magdeburg" },
-    { value: "ns", label: "Novi Sad" },
+  { value: "", label: "- choose option -" },
+  { value: "hd", label: "Heidelberg" },
+  { value: "md", label: "Magdeburg" },
+  { value: "ns", label: "Novi Sad" },
 ];
 
 const DeskContainer = () => {
-    const [location, setLocation] = useState("");
-
-    const handleLocationChange = (event) => {
-        setLocation(event.target.value);
-        setSelectedDesk(null);
-        console.log(`location changed to ${event.target.value}`);
-    };
+  const [location, setLocation] = useState("");
+  const [desks, setDesks] = useState([]);
+  const [selectedDesk, setSelectedDesk] = useState(null);
 
-    //selected desk variable to change on button clicked
-    const [selectedDesk, setSelectedDesk] = useState(null);
+  const handleLocationChange = (event) => {
+    setLocation(event.target.value);
+    setSelectedDesk(null);
+    console.log(`Location changed to ${event.target.value}`);
+  };
 
-    useEffect(() => {
-        if (selectedDesk !== null) {
-            console.log(selectedDesk);
-        }
-    }, [selectedDesk]);
+  useEffect(() => {
+    handleDesks();
```

Comment:
I get this warning on `location` in the `useEffect` dependency array: "React Hook useEffect has a missing dependency: 'handleDesks'. Either include it or remove the dependency array.eslintreact".

See my commit [here](https://github.com/NikolaVetnic/js_desks-app/pull/12/commits/2912b01a4970d2ce660d205faca2c66d1dd414c5) as to how I solved it - basically, I used the `useCallback` hook.